### PR TITLE
[docs] Add zstd and ninja to version ranges supported

### PR DESCRIPTION
**Pull request by uilianries**
#### Motivation

Both `zstd` and `ninja` looks pretty stable and should be backward compatible to be used in ConanCenterIndex.

#### Details

- Zstd: 
  - https://abi-laboratory.pro/index.php?view=timeline&l=zstd
  - https://github.com/facebook/zstd/issues/731#issuecomment-310264043
  
  Zstd 1.5 is supported since 2023, so keeping 1.5.x is safest, just in case.

- Ninja
  - https://ninja-build.org/manual.html#ref_versioning

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by jcar87 — DISMISSED** [review]

**Review by AbrilRBS — DISMISSED** [review]

File: docs/adding_packages/dependencies.md [hunk 1]
```diff
@@ -192,6 +192,8 @@ version range only when a requirement for a newer version is needed.
 * qt5: `[~5.15]`, if your library depends on qt5, only the 5.15 minor version is allowed
 * qt6: `[>=6.x <7]`, where 6.x is the lower bound of your needed qt6 version
 * c-ares: `[>=1.27 <2]`
+* zstd: `[^1.5]` it's equivalent to `[>=1.5 <1.6]`
+* ninja: `[>=1.10.2 <2]`
```
Comment (uilianries) [inline docs/adding_packages/dependencies.md:196]:
```suggestion
* ninja: `[>=1.10.2 <2]`
* meson: `[>=1.2.3 <2]`
```

**Comment by uilianries** [comment]
I just included Meson too. It looks stable so far. The version 1.2.x is well used in CCI and is 1 year old.

**Review by AbrilRBS — APPROVED** [review]

**Comment by valgur** [comment]
Maybe add `pkgconf/[>=2.2 <3]` to the list as well, based on https://github.com/conan-io/conan-center-index/commit/be88746d26de374e2b0b483553a8c78a04c57996#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R405 ?

File: docs/adding_packages/dependencies.md [hunk 2]
```diff
@@ -192,6 +192,9 @@ version range only when a requirement for a newer version is needed.
...
+* meson: `[>=1.2.3 <2]`
```
Comment (jcar87) [inline docs/adding_packages/dependencies.md:197]:
```suggestion
* meson: `[>=1.2.3 <2]`
* pkgconf: `[>=2.2 <3]`
```

**Review by jcar87 — APPROVED** [review]

**Review by AbrilRBS — APPROVED** [review]

File: docs/adding_packages/dependencies.md [hunk 3]
```diff
@@ -192,6 +196,10 @@ version range only when a requirement for a newer version is needed.
 * qt5: `[~5.15]`, if your library depends on qt5, only the 5.15 minor version is allowed
 * qt6: `[>=6.x <7]`, where 6.x is the lower bound of your needed qt6 version
 * c-ares: `[>=1.27 <2]`
+* zstd: `[^1.5]` it's equivalent to `[>=1.5 <1.6]`
```
Comment (SpaceIm) [inline docs/adding_packages/dependencies.md:199]: AFAIK, `[^1.5]` it's equivalent to `[>=1.5 <2]`

Comment (uilianries) [reply docs/adding_packages/dependencies.md:199]: You are right. https://docs.conan.io/1/versioning/version_ranges.html#version-ranges I'll open a PR fixing it. Thank you!

**Pull request merged**